src/traces/violin/calc.js

```diff
@@ -0,0 +1,75 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var Lib = require('../../lib');
+var boxCalc = require('../box/calc');
+
+var kernels = {
+    gaussian: function(v) {
+        return (1 / Math.sqrt(2 * Math.PI)) * Math.exp(-0.5 * v * v);
```

Comment:
We could support more:

https://en.wikipedia.org/wiki/Kernel_(statistics)#Kernel_functions_in_common_use

Reply:
Good thinking to make this extensible down the line. There would be something nice about using one of the polynomial kernels that goes smoothly to zero at a finite position but to start just gaussian seems fine. That's the ggplot2 default anyway, and I can't find anything to say which one seaborn uses.

Reply:
I added the _epanechnikov_ kernel in https://github.com/plotly/plotly.js/pull/2116/commits/ad51966b2c72b59633415f711349b6836a9cf3e8

Reply:
... and of course, I could add a few more if desired. Note that other kernels make the violins look a little less smooth than the gaussian. Perhaps this is why other libraries (e.g. seaborn and ggplot) only use the gaussian kernel for violin plots :thinking: